javascript/express/security/injection/tainted-sql-string.yaml

```diff
@@ -20,34 +21,56 @@ rules:
       category: security
       technology:
         - express
+      license: Commons Clause License Condition v1.0[LGPL-2.1-only]
     languages:
       - javascript
       - typescript
     mode: taint
     pattern-sources:
       - patterns:
-          - pattern-inside: |
-              require('express');
-              ...
           - pattern-either:
               - pattern-inside: function ... ($REQ, $RES) {...}
               - pattern-inside: function ... ($REQ, $RES, $NEXT) {...}
-          - pattern: $REQ
+              - patterns:
+                  - pattern-inside: $APP.$METHOD(..., function $FUNC($REQ, $RES) {...})
+                  - metavariable-regex:
+                      metavariable: $METHOD
+                      regex: ^(get|post|put|head|delete|options)
+          - pattern-either:
+              - pattern: $REQ.query
+              - pattern: $REQ.body
+              - pattern: $REQ.params
+              - pattern: $REQ.cookies
+              - pattern: $REQ.headers
+      - patterns:
+          - pattern-either:
+              - pattern-inside: >
+                  ({ $REQ }: Request,$RES: Response, $NEXT: NextFunction) =>
+                  {...}
+              - pattern-inside: |
+                  ({ $REQ }: Request,$RES: Response) => {...}
+          - pattern-either:
+              - pattern: params
+              - pattern: query
+              - pattern: cookies
+              - pattern: headers
+              - pattern: body
     pattern-sinks:
       - patterns:
           - pattern-either:
               - patterns:
                   - pattern-either:
-                      - pattern: |
+                      - pattern-inside: |
                           "$SQLSTR" + $EXPR
-                      - pattern: |
-                          "$SQLSTR".concat(...)
-                      - pattern: util.format($SQLSTR, ...)
+                      - pattern-inside: |
+                          "$SQLSTR".concat($EXPR)
+                      - pattern: util.format($SQLSTR, $EXPR)
```

Comment:
It's strange to mix `pattern-inside` and `pattern` in a `pattern-either`. Is that what you want?